crates/precompile/src/blob/kzg_settings.rs

```diff
@@ -0,0 +1,34 @@
+use alloc::boxed::Box;
+use c_kzg::KzgSettings;
+use once_cell::race::OnceBox;
+
+#[rustfmt::skip]
+mod generated;
+
+/// Note: the type of this is an implementation detail, so we don't expose it in the public API.
+static GLOBAL_KZG_SETTINGS: OnceBox<KzgSettings> = OnceBox::new();
+
+/// Returns a reference to the global [`KzgSettings`] instance, initializing it with the default
+/// value if it was not previously set.
+pub fn get_global_or_default() -> &'static KzgSettings {
+    GLOBAL_KZG_SETTINGS.get_or_init(|| Box::new(default()))
```

Comment:
> which should be fine, but it would mean that it is impossible to set the global kzg after a transact call.

Hmm does this mean that you cannot test w/ various parameters in a loop? I guess it might be a fine limitation agree about doc